util/compare_test_results.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python3
+"""
+Compare GNU test results between current run and reference to identify
+regressions and fixes.
+
+
+Arguments:
+    CURRENT_JSON       Path to the current run's aggregated results JSON file
+    REFERENCE_JSON     Path to the reference (main branch) aggregated
+                        results JSON file
+    --ignore-file      Path to file containing list of tests to ignore
+                        (for intermittent issues)
+    --output           Path to output file for GitHub comment content
+"""
+
+import argparse
+import json
+import os
+import sys
+
+
+def flatten_test_results(results):
+    """Convert nested JSON test results to a flat dictionary of test paths to statuses."""
+    flattened = {}
+    for util, tests in results.items():
+        for test_name, status in tests.items():
+            test_path = f"{util}/{test_name}"
+            flattened[test_path] = status
+    return flattened
+
+
+def load_ignore_list(ignore_file):
+    """Load list of tests to ignore from file."""
+    if not os.path.exists(ignore_file):
+        return set()
+
+    with open(ignore_file, "r") as f:
+        return {line.strip() for line in f if line.strip() and not line.startswith("#")}
+
+
+def identify_test_changes(current_flat, reference_flat):
+    """
+    Identify different categories of test changes between current and reference results.
+
+    Args:
+        current_flat (dict): Flattened dictionary of current test results
+        reference_flat (dict): Flattened dictionary of reference test results
+
+    Returns:
+        tuple: Four lists containing regressions, fixes, newly_skipped, and newly_passing tests
+    """
+    # Find regressions (tests that were passing but now failing)
+    regressions = []
+    for test_path, status in current_flat.items():
+        if status == "FAIL" or status == "ERROR":
+            if test_path in reference_flat:
+                if (
+                    reference_flat[test_path] == "PASS"
+                    or reference_flat[test_path] == "SKIP"
+                ):
```

Comment:
I think you can simplify it to:
```suggestion
                if reference_flat[test_path] in ("PASS", "SKIP"):

```